### DISCUSSION TITLE
[Android]Dynamic Link not trigger on Fresh App Install?

### DISCUSSION THREAD

**Discussion by kkusanagi** [invertase/react-native-firebase#4139] [Q&A]
for `react-native-firebase 5.6`, I'm able to trigger the `getInitialLink` when the app is freshly installed.

but after use `@react-native-firebase/dynamic-links@7.4.3`, it failed to trigger the `getInitialLink` when fresh install.

Anyone have this issue? or just me?

Edited:
This is using `buildShortLink`. Or Had I set something wrong?

```
const url = await DynamicLink().buildShortLink({
      link: httpurl,
      domainUriPrefix: 'https://'+headerlink,
      android:{
        packageName:'com.example.example'
      },
      ios:{
        bundleId: 'com.example.example',
        appStoreId: '[APPID]'
      }
    },"SHORT");
```

**Comment by Salakar** [comment]
Is this iOS, Android?

**Reply by kkusanagi** [reply]
Title state Android. So is Android.

**Comment by Bbw0706** [comment]
I have same issue on IOS too, the issue only occurs when first installed and open the apps

**Comment by kkusanagi** [comment]
Hey, ummm, it's been a while now. And I really need this features to do sharing for membership system.
Had tried use REST API on firebase to generate and result still same. Is there anything we need to change.

This only happen for fresh installed app. The installed app have no issue for this.

**Comment by mikehardy** [comment]
1. The title does not actually say android or ios. So that was a pertinent question. I understand you @kkusanagi have a problem with iOS. @Bbw0706 has "the same problem" (note: no problem is ever the same, really) on android.
1. "getInitialLink" is not triggered. "onLink" is triggered. "getInitialLink" is something you call.

This would be better suited to a full bug report where we can see details. For instance, what version of ios pods are you using? Is this on a simulator or real device? (it's my understanding it needs to be on a real device for representative testing of link). What version of iOS is affected? What exactly is the code doing? It would be useful to see an App.js that showed a full reproduction

Can you reproduce with the official dynamic links quickstart? https://github.com/firebase/quickstart-ios/tree/master/dynamiclinks

**Reply by kkusanagi** [reply]
Haha, I really mess up with the settings in Android. Luckily found some solution from Stackoverflow and old react-native-firebase 5.6.0 documentation.

Seems like the Android need to update some data in AndroidManifest.xml 

Need to change the 
`android:launchMode="singleTask"`
and Add new line for new link. Not replace them.
```
            <data android:host="example2.page.link" android:scheme="http"/>
            <data android:host="example2.page.link" android:scheme="https"/>
```

```
<activity
        android:name=".MainActivity"
        android:label="@string/app_name"
        android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
        android:windowSoftInputMode="adjustResize"
        android:launchMode="singleTask">
        <intent-filter>
            <action android:name="android.intent.action.VIEW"/>
            <category android:name="android.intent.category.DEFAULT"/>
            <category android:name="android.intent.category.BROWSABLE"/>
            <data android:host="example.page.link" android:scheme="http"/>
            <data android:host="example.page.link" android:scheme="https"/>
            <data android:host="example2.page.link" android:scheme="http"/>
            <data android:host="example2.page.link" android:scheme="https"/>
        </intent-filter>
      </activity>
```
